lib/Eventz.php

```diff
@@ -393,6 +396,64 @@ public static function get_event_dates( $event ) {
         return $dates;
     }
 
+    /**
+     * Get event entries info
+     * Entries occur if the event is held weekly for specific days
+     *
+     * @param object $event Event object.
+     *
+     * @return array
+     */
+    public static function get_event_entries( $event ) {
+        $entries    = [];
+        $entry_data = [];
+
+        if ( empty( $event->event->entries ) ) {
+            return $entries;
+        }
+
+        // Get single entry data
+        foreach ( $event->event->entries as $entry ) {
+            $entry_data[] = [
+                'day_of_week' => $entry->dayOfWeek,
+                'start_time'  => $entry->startTimeLocal,
+                'end_time'    => $entry->endTimeLocal,
+                'sold_out'    => $entry->isSoldOut,
+            ];
+        }
+
+        $start_date = \DateTime::createFromFormat( 'Y-m-d H:i:s', date( 'Y-m-d H:i:s', strtotime( $event->event->start ) ) );
+        $end_date   = \DateTime::createFromFormat( 'Y-m-d H:i:s', date( 'Y-m-d H:i:s', strtotime( $event->event->end ) ) );
+
+        // Loop through days and get the dates each week
+        foreach ( $entry_data as $entry ) {
+            if ( date( 'D', strtotime( $entry['day_of_week'] ) ) != $start_date->format( 'D' ) ) {
+                $day_of_week = date( 'D', strtotime( $entry['day_of_week'] ) );
+                $start_date->modify( "next $day_of_week" );
+            }
+
+            // Get all occurences of day in the time range
+            while ( $start_date <= $end_date ) {
```

Comment:
Varmaan tarkoitus verrata if:llä, eikä while loopilla 🤔
```suggestion
            if ( $start_date <= $end_date ) {
```

Reply:
While on ihan tarkoituksellinen. Tässä halutaan loopata $start_datea niin kauan kuin se on pienempi tai sama kuin $end_date. Joka iteraatiossa $start_date siirretään viikko eteenpäin.